packages/sanity/src/core/studio/colorSchemeStore.ts

```diff
@@ -0,0 +1,39 @@
+import {type StudioThemeColorSchemeKey} from '../theme/types'
+
+function getScheme(scheme: unknown): StudioThemeColorSchemeKey {
+  switch (scheme) {
+    case 'dark':
+    case 'light':
+      return scheme
+    default:
+      return 'system'
+  }
+}
+
+/** @internal */
+export const LOCAL_STORAGE_KEY = 'sanityStudio:ui:colorScheme'
+
+let snapshot: StudioThemeColorSchemeKey
```

Comment:
React Compiler doesn't like writing to ESM bindings like this one, during render.
That's fine, we can move it into a separate file like this and the compiler no longer complains, as it's out of scope :)